### ISSUE TITLE
Expose highlighter logic

### ISSUE THREAD

**Issue by bebraw** [bebraw/reactabular#36]
On second thought, does exposing the logic do any good? After all you'll want to show results that matched. If highlighting logic is different than search logic, then you could end up with a table of search results without any highlights. Makes sense?

CC @eviltoylet

**Comment by eviltoylet** [comment]
Well, it gets tricky. For example, if the search component is not matching case sensitivity, then the highlighter should know that it shouldn't match case sensitivity on both the query and the target to highlight. The same issue happens with the type of matching, e.g. infix, prefix, suffix. The goal is to encapsulate this without needing to pass the same data into two separate components.

I think the best thing to do is the following:
- have the Search component expose a method called `matches(column, value)`
- Return a structure of the following format which the highlighter can use to highlight all the start/end index chunks.
```
    matchIndexes: [
       {
           startIndex: ,
           endIndex:
       }
    ]
```

**Comment by bebraw** [comment]
I see, that dependency is definitely a problem. As you stated, having a way to pipe search results with matches from Search component could be one way.

Note that things get a little interesting when you perform a search over backend. In that case we would have to do something different. In a case of mine the search endpoint returns whole rows paginated. For highlighting to work I will have to traverse through the data anyway.

I suppose the interface you are proposing could work here. In the backend case I need to go through all rows, perform logic to generate the structure highlight would expect and pass it there.

In short highlight component would deal just with rendering. We would effectively extract the logic part out of it making things more flexible. That's good.
